### PR TITLE
Added Azure blob example

### PR DESCRIPTION
**Pull request by gaurav-nelson**
Added example for configuring Azure Blob Storage for OpenShift integrated docker registry.

### CODE REVIEW

**Comment by gaurav-nelson** [comment]
@tanaka-takayoshi Can you please review this as well?

**Comment by tanaka-takayoshi** [comment]
Thanks @gaurav-nelson , it looks good to me.

**Comment by gaurav-nelson** [comment]
@bfallonf PTAL

**Comment by vikram-redhat** [comment]
@openshift/team-documentation - PTAL.

@gaurav-nelson - what labels are required for this?

File: install_config/storage_examples/azure_blob_docker_registry_example.adoc [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+. Set the `REGISTRY_CONFIGURATION_PATH` environment variable:
++
+[source,bash]
+----
+$ oc set env dc/docker-registry \
+    REGISTRY_CONFIGURATION_PATH=/etc/docker/registry/config.yaml
+----
+
+[NOTE]
+====
```
Comment (ahardin-rh) [inline install_config/storage_examples/azure_blob_docker_registry_example.adoc]: I would remove this from a Note box

File: install_config/storage_examples/azure_blob_docker_registry_example.adoc [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+[[azure-blob-docker-registry-prerequisites]]
+== Before you begin
+
+* *Create a storage container in a blob service in a storage account*: Create a storage container using Microsoft Azure Portal, Microsoft Azure CLI, or Microsoft Azure Storage Explorer. Keep a note of the *_storage account name_*, *_storage account key_* and *_container name_*.
+* *Deploy the integrated Docker registry*: link:https://docs.openshift.com/container-platform/latest/install_config/registry/deploy_registry_existing_clusters.html#deploy-registry[Deploy the registry] if it is not deployed.
+
+[[azure-blob-docker-registry-registry-config]]
+== Overriding Registry Configuration
+
+Following steps creates a new registry pod which replaces the old pod automatically.
```
Comment (ahardin-rh) [inline install_config/storage_examples/azure_blob_docker_registry_example.adoc]: To create a new registry pod and replace the old pod automatically:

File: install_config/storage_examples/azure_blob_docker_registry_example.adoc [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+:prewrap:
+
+toc::[]
+
+[[azure-blob-docker-registry-overview]]
+== Overview
+This topic reviews how to configure Azure Blob Storage for OpenShift integrated Docker Registry.
+
+[[azure-blob-docker-registry-prerequisites]]
+== Before you begin
```
Comment (ahardin-rh) [inline install_config/storage_examples/azure_blob_docker_registry_example.adoc]: Before You Begin

File: install_config/storage_examples/azure_blob_docker_registry_example.adoc [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+:data-uri:
+:icons:
+:experimental:
+:toc: macro
+:toc-title:
+:prewrap:
+
+toc::[]
+
+[[azure-blob-docker-registry-overview]]
```
Comment (ahardin-rh) [inline install_config/storage_examples/azure_blob_docker_registry_example.adoc]: We typically do not use IDs for Overview sections

File: install_config/storage_examples/azure_blob_docker_registry_example.adoc [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+== Overview
+This topic reviews how to configure Azure Blob Storage for OpenShift integrated Docker Registry.
```
Comment (ahardin-rh) [inline install_config/storage_examples/azure_blob_docker_registry_example.adoc]: Microsoft Azure Blob storage for the OpenShift-integrated Docker Registry (or do you mean [OpenShift Container Registry](https://doc-stage.usersys.redhat.com/documentation/en-us/ccs_internal_documentation/1.0/html-single/glossary_of_terms_and_conventions_for_product_documentation/#openshift-container-registry)?)

File: install_config/storage_examples/azure_blob_docker_registry_example.adoc [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+
+[[azure-blob-docker-registry-prerequisites]]
+== Before you begin
+
+* *Create a storage container in a blob service in a storage account*: Create a storage container using Microsoft Azure Portal, Microsoft Azure CLI, or Microsoft Azure Storage Explorer. Keep a note of the *_storage account name_*, *_storage account key_* and *_container name_*.
```
Comment (ahardin-rh) [inline install_config/storage_examples/azure_blob_docker_registry_example.adoc]: Maybe just make these steps and take out the redundancy of the bolded parts?

File: install_config/storage_examples/azure_blob_docker_registry_example.adoc [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+* *Deploy the integrated Docker registry*: link:https://docs.openshift.com/container-platform/latest/install_config/registry/deploy_registry_existing_clusters.html#deploy-registry[Deploy the registry] if it is not deployed.
+
+[[azure-blob-docker-registry-registry-config]]
+== Overriding Registry Configuration
+
+Following steps creates a new registry pod which replaces the old pod automatically.
+
+. Create a new registry configuration file called `registryconfig.yaml` and add following information:
```
Comment (ahardin-rh) [inline install_config/storage_examples/azure_blob_docker_registry_example.adoc]: *_registryconfig.yaml_*

Comment (ahardin-rh) [inline install_config/storage_examples/azure_blob_docker_registry_example.adoc]: add the following information:

File: install_config/storage_examples/azure_blob_docker_registry_example.adoc [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+
+[[azure-blob-docker-registry-overview]]
+== Overview
+This topic reviews how to configure Azure Blob Storage for OpenShift integrated Docker Registry.
+
+[[azure-blob-docker-registry-prerequisites]]
+== Before you begin
+
+* *Create a storage container in a blob service in a storage account*: Create a storage container using Microsoft Azure Portal, Microsoft Azure CLI, or Microsoft Azure Storage Explorer. Keep a note of the *_storage account name_*, *_storage account key_* and *_container name_*.
+* *Deploy the integrated Docker registry*: link:https://docs.openshift.com/container-platform/latest/install_config/registry/deploy_registry_existing_clusters.html#deploy-registry[Deploy the registry] if it is not deployed.
```
Comment (mburke5678) [inline install_config/storage_examples/azure_blob_docker_registry_example.adoc]: Should this be a relative xref rather than a hard link?

**Comment by mburke5678** [comment]
@gaurav-nelson One nit

File: install_config/storage_examples/azure_blob_docker_registry_example.adoc [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+          pullthrough: true
+          enforcequota: false
+          projectcachettl: 1m
+          blobrepositorycachettl: 10m
+    storage:
+      - name: openshift
+----
+<1> Replace the values for `accountname`, `acountkey`, and `container` with *_storage account name_*, *_storage account key_*, and *_storage container name_* respectively.
+
+. Run the following command to create a new registry configuration:
```
Comment (bfallonf) [inline install_config/storage_examples/azure_blob_docker_registry_example.adoc]: Similar to comments above, maybe just "Create a new registry configuration:". The front bits of the sentence are superfluous.

File: install_config/storage_examples/azure_blob_docker_registry_example.adoc (hunk 5 above)
Comment (gaurav-nelson) [inline install_config/storage_examples/azure_blob_docker_registry_example.adoc]: In the section which I link to it is listed as 'Integrated Docker registry' https://docs.openshift.com/container-platform/latest/install_config/registry/deploy_registry_existing_clusters.html#deploy-registry

**Comment by gaurav-nelson** [comment]
Thanks @ahardin-rh  @bfallonf  @mburke5678 I have addressed all issues in the latest commit.

**Comment by gaurav-nelson** [comment]
@mburke5678 Can you please check and merge this one?

File: install_config/storage_examples/azure_blob_docker_registry_example.adoc [hunk 10]
```diff
@@ -0,0 +1,108 @@
...
+:toc-title:
+:prewrap:
+
+toc::[]
+
+== Overview
+This topic reviews how to configure Azure Blob Storage for OpenShift integrated Docker Registry.
+
+[[azure-blob-docker-registry-prerequisites]]
+== Before you Begin
```
Comment (mburke5678) [inline install_config/storage_examples/azure_blob_docker_registry_example.adoc]: Before You Begin (cap Y on You)

File: install_config/storage_examples/azure_blob_docker_registry_example.adoc [hunk 11]
```diff
@@ -0,0 +1,108 @@
...
+:icons:
+:experimental:
+:toc: macro
+:toc-title:
+:prewrap:
+
+toc::[]
+
+== Overview
+This topic reviews how to configure Azure Blob Storage for OpenShift integrated Docker Registry.
```
Comment (mburke5678) [inline install_config/storage_examples/azure_blob_docker_registry_example.adoc]: Do you want a link here to the registry docs.
**https://docs.openshift.com/container-platform/latest/install_config/registry/deploy_registry_existing_clusters.html#deploy-registry**

File: install_config/storage_examples/azure_blob_docker_registry_example.adoc (hunk 9 above)
Comment (mburke5678) [inline install_config/storage_examples/azure_blob_docker_registry_example.adoc]: @gaurav-nelson Did you want to take the paragraph out as @bfallonf and @ahardin-rh suggested?

**Comment by mburke5678** [comment]
@gaurav-nelson Couple of comments. Otherwise LGTM

**Comment by gaurav-nelson** [comment]
[rev_history]
|xref:../install_config/storage_examples/azure_blob_docker_registry_example.adoc#azure-blob-docker-registry[Configuring Azure Blob Storage for Integrated Docker Registry]
|Added an example for configuring Azure Blob Storage for OpenShift Integrated Docker Registry.
%

**Pull request merged**